docs/user_guide/concepts/main_concepts/data_management.rst

```diff
@@ -9,69 +9,65 @@ Understand How Flyte Handles Data
 Types of Data
 =============
 
-There are two parts to the data in Flyte:
+In Flyte, data is categorized into metadata and raw data to optimize data handling and improve performance and security.
 
-1. Metadata
+* **Metadata**: Small values, like integers and strings, are treated as "stack parameters" (passed by value). This metadata is globally accessible to Flyte components (FlytePropeller, FlyteAdmin, and other running pods/jobs). Each entry is limited to 10MB and is passed directly between tasks. On top of that, metadata allow in-memory computations for branches, partial outputs, and composition of multiple outputs as input for other tasks.
```

Comment:
> `Each entry is limited to 10MB and is passed directly between tasks.`

For future proofing of this section, shouldn't we add here that bigger files are automatically offloaded?
cc @pingsutw